### PR TITLE
Add a tool to run multiple YAML cases locally

### LINKED ISSUE #2989: Develop a tool to manually run CI tests
### What new functionality do you need?

The `create_experiment.py` creates an experiment from a YAML and sources all of the appropriate configuration files.  To run this script, modules must be loaded and environmental variables (`RUNDIRS` and `pslot`) must be set.  It would be handy to add a tool to wrap around `create_experiment.py` that handles the variable creation, module loading, pointing to the correct YAML, etc.  Additionally, it could append the experiments to the user's crontab (or scrontab on Gaea) to automatically start running experiments.

### What are the requirements for the new functionality?

A simple user-interfaced command line tool that takes
1) the location where the experiment(s) will run (i.e. `RUNDIRS`)
2) a list of valid tests held in `ci/cases/pr` (or possibly other locations)
3) a flag indicating if the crontab should be updated
4) an optional email argument to write to the crontab

### Acceptance Criteria

- [ ] Invalid `RUNDIRS` argument will raise an error
- [ ] Valid test cases are runnable
- [ ] Invalid test cases raise an error
- [ ] Crontab is updated when requested

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# Description
This adds the script `generate_workflows.py`, which provides a flexible, multipurpose feature set to run suites of cases from YAMLs with minimal intervention.  For most cases, this will allow users to easily run CI tests on a local platform, including updating the user's crontab automatically.

Resolves #2989

# Type of change
- [x] New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES
- Does this change require an update to any submodules? NO

# How has this been tested?
- [ ] Run all CI test cases on Hera and WCOSS2
- [ ] Verify that all switches work as expected in combination

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] I have made corresponding changes to the system documentation if necessary

### CODE REVIEW

**Comment by aerorahul** [comment]
Not opposed to this feature, and if users wish to run all ci tests in one shot, more power to them.
I won't have time to review this in depth before I leave, so I will leave with one comment.
Just assign yourself (@DavidHuber-NOAA) to this script in the `CODEOWNERS` file, so that you get notified (in addition to the rest) when there is a change to this script.

**Review by aerorahul — DISMISSED** [review]
approve this feature.
Just one request to add @DavidHuber-NOAA to this script in the `CODEOWNERS` file so that he is notified of any changes to this script.

File: workflow/generate_workflows.sh [hunk 1]
```diff
@@ -0,0 +1,532 @@
+#!/usr/bin/env bash
+
+###
+function _usage() {
+   cat <<-EOF
```
Comment (WalterKolczynski-NOAA) [inline workflow/generate_workflows.sh:5]: `<<-` won't work as expected without using tabs for indenting
```suggestion
   cat << EOF
```

File: workflow/generate_workflows.sh [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+      echo "Rerun with -h for usage examples."
+      exit 4
+   fi
+   set -e
+else
+   echo "The RUNTESTS directory ${_runtests} already exists."
+   echo "Would you like to remove it?"
+   _attempts=0
+   while read -r _from_stdin
+   do
```
Comment (WalterKolczynski-NOAA) [inline workflow/generate_workflows.sh:208]:
```suggestion
   while read -r _from_stdin; do
```

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**